### PR TITLE
Programmatically generate the menubar from config

### PR DESCRIPTION
**Pull request by SteveMarshall**
The current menubar is easy to understand, but isn't easy to make show the currently active page. It also requires some duplication, making future markup changes more complicated.

This change moves the control of the navigation structure into the `_config.yml` (though it could possibly be done another way), and then uses that structure to generate the menu programmatically. Menu items are rendered in the order they're stored in the array, and may either be the ID of a page, or a dictionary (expected to contain both `title` and `url`, which we output without touching).

The rendering code checks if the menu item is either the current page, or is mentioned in the current page's `has-nav` attribute and, if true, marks the menu item as selected.

### CODE REVIEW

**Comment by annashipman** [comment]
It would be helpful to be able to show the currently selected page and reduce markup duplication, thanks 👍 

Please could you fix these two issues:

1. The build is failing because `/scripts/myproposal.php` doen't exist in this repo; it needs to be a fully qualified URL as per the menu item it replaces.

2. Please can you also update the README where it explains how to add/remove an item from the menu?

Suggestions:

1. It might be worth adding some comments to the code in `_includes/menu.html` because I found it a little hard to follow, and users of this site will be new programme chairs every year (so needs to be easy for people to understand first time).

2. There was a page commented out in the menu which we know is going to be added back in; is it worth (/possible to) propogating that through to this implementation?

**Comment by SteveMarshall** [comment]
@annashipman I've fixed all of those (I think?), and pushed a fixup commit. Do you want me to change the title of `sponsor.md` to `Sponsor us`, so the navigation is consistent with how it was before, or should we leave it as `Sponsorship opportunities`?

**Comment by annashipman** [comment]
I think 'Sponsor us' is better, please.

The rest of it is great, 👍. Happy for you to squash the commits now I've seen the changes.
